.github/workflows/build_libs.yml

```diff
@@ -104,35 +92,37 @@ jobs:
       matrix:
         lib: ${{ fromJson(needs.changes.outputs.libs) }}
         build_type: ['Release', 'Debug']
-        shared: ['True', 'False']
-    env:
-      CONAN_REVISIONS_ENABLED: 1
     steps:
     - uses: actions/checkout@v3
-    - name: install conan
-      run: pip install conan~=1.53
+    - name: Install conan
+      uses: turtlebrowser/get-conan@v1.2
+      with:
+        version: '2.0.2'
     - name: create profile
-      run: conan profile new tket --detect --force
-    - name: add remote
+      run: conan profile detect --force
+    - name: set remotes
       run: |
-        conan remote clean
+        rm `conan config home`/remotes.json
         conan remote add tket-libs https://quantinuumsw.jfrog.io/artifactory/api/conan/tket1-libs
     - name: build ${{ matrix.lib }}
-      run: conan create --profile=tket -s build_type=${{ matrix.build_type }} -o ${{ matrix.lib }}:shared=${{ matrix.shared }} libs/${{ matrix.lib }} tket/stable --build=missing
+      run: conan create -s build_type=${{ matrix.build_type }} -o boost/*:header_only=True libs/${{ matrix.lib }} --build=missing --user=tket --channel=stable --format json > ${{ matrix.lib }}.json
     - name: authenticate to repository
-      run: conan user -p ${{ secrets.JFROG_ARTIFACTORY_TOKEN_2 }} -r tket-libs ${{ secrets.JFROG_ARTIFACTORY_USER_2 }}
+      run: conan remote login -p ${{ secrets.JFROG_ARTIFACTORY_TOKEN_2 }} tket-libs ${{ secrets.JFROG_ARTIFACTORY_USER_2 }}
     - name: get version
       run: |
-        lib_ver=$(conan inspect --raw version libs/${{ matrix.lib }}/conanfile.py)
-        echo "LIB_VER=${lib_ver}" >> $GITHUB_ENV
-    - name: upload package (dry run)
-      if: github.event_name == 'pull_request'
-      run: conan upload ${{ matrix.lib }}/${LIB_VER}@tket/stable --all -r=tket-libs --skip-upload
+        lib_label=`./libver ${{ matrix.lib }}.json ${{ matrix.lib }}`
+        echo "LIB_LABEL=${lib_label}" >> $GITHUB_ENV
     - name: upload package
       if: github.event_name == 'push'
-      run: conan upload ${{ matrix.lib }}/${LIB_VER}@tket/stable --all -r=tket-libs
+      run: conan upload ${{ env.LIB_LABEL }} -r=tket-libs
     - name: unauthenticate
-      run: conan user -c
+      if: always()
```

Comment:
is always() a good condition?

Reply:
Yes, this ensures it is always executed even when a step fails.